server/search-index.js

```diff
@@ -11,42 +10,38 @@ const {
 } = require("../content");
 const { SearchIndex } = require("../build");
 
-const promisifed = util.promisify(fs.readFile);
-
 function* walker(root) {
-  const files = fs.readdirSync(root);
-  for (const name of files) {
-    const filepath = path.join(root, name);
-    const isDirectory = fs.statSync(filepath).isDirectory();
-    if (isDirectory) {
+  const dirents = fs.readdirSync(root, { withFileTypes: true });
+  for (const dirent of dirents) {
+    // Doing it this way is faster than doing `path.join(root, dirent.name)`
+    const filepath = `${root}${path.sep}${dirent.name}`;
```

Comment:
Faster but probably less safe (https://github.com/nodejs/node/blob/master/lib/path.js#L389).

Reply:
Why less safe?
I can imagine the `path.join(DIR1, DIR2)` is "safer" if there's a chance that `DIR1` *might* end with a `/` (or `path.sep`). But in this context, we always know that `root` does NOT end with a `/`.